plugin/normalize.go

```diff
@@ -63,8 +63,34 @@ type (
 
 // Normalize will return the host portion of host, stripping
 // of any port or transport. The host will also be fully qualified and lowercased.
+// An empty string is returned on failure
+// Deprecated: use OriginsFromArgsOrServerBlock or NormalizeExact
+func (h Host) Normalize() string {
+	// The error can be ignored here, because this function should only be called after the corefile has already been vetted.
```

Comment:
It would be nice to emit a warning, ideally at build time but since these are only called in setup, I would find runtime acceptable too.

Reply:
I believe `golint` will flag the use of deprecated functions (identified by "Deprecated:" in function comments).  But we don't automate code linting.

I can add a runtime warning.